search/tests/mock_search_engine.py

```diff
@@ -343,9 +380,15 @@ def score_documents(documents_to_search):
             kwargs["from_"] if "from_" in kwargs else None,
             sorted(search_results, key=lambda k: k["score"])
         )
-        return {
+
+        response = {
             "took": 10,
             "total": len(search_results),
             "max_score": max_score,
             "results": results
         }
+
+        if facet_terms:
+            response["facets"] = _process_facet_terms(documents_to_search, facet_terms)
```

Comment:
Is _process_facet_terms along the lines of a Command pattern implementation?  Fine if it is, but as noted above it would be good to describe the processes taking place in more detail, in the method's docstring.